hls4ml/model/optimizer/passes/infer_precision.py

```diff
@@ -561,15 +562,34 @@ def _infer_rnn_precision(self, node, types_to_infer):
 
         return inferred_types
 
-    def _infer_par_act_precision(self, node, types_to_infer):
+    def _infer_const_precision(self, node, type_to_infer, attr_name):
         inferred_types = []
 
-        # For threshold relu, set the parameter precision to be the input precision by default;
-        # for other parametrized activations, just allow the default precision to be used.
-        # Can override these values in the configuration by explicitly setting them.
-        if 'param_t' in inferred_types and self.get_attr('activation').lower() == 'thresholdedrelu':
-            in_type = node.get_input_variable().type.precision
-            node.attributes['param_t'].type = in_type
-            inferred_types.append('param_t')
+        def get_man_exp(f):
+            f = np.abs(f)
+            s = struct.pack('>f', f)
```

Comment:
Use calculations:  based on the value, you can easily determine how many bits you need. Going to structs is hard to follow.